build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -413,7 +415,7 @@ export function changePreReleaseIdentifier(
 	}
 
 	const pr = ver.prerelease;
-	if (pr.length < 1) {
+	if (pr.length === 0) {
 		throw new Error(`Version has no prerelease section: ${version}`);
```

Comment:
This seems odd to me given that there was another change that made `.length !== 0` become `.length > 0`. I never expect length to be less than 0, but it is unsigned; so, I like the prior check more. Also think `.length <= 0` or `!(foo.length > 0)` are more in spirit. Why is `=== 0` preferred here?
(This *is* fine, I would just like to understand.)

Reply:
This is just following the linter preferences, right? My guess for `=== 0` and `> 0` is that it considers them the cleaner analogous versions of "is empty" and "is not empty", over "has less than 1 item" and "has some non-zero (implicitly non-negative) number of items". I could see `!== 0` instead of `> 0`, but don't have strong feelings about it.

Reply:
Yeah, matches my understanding.

Reply:
I also don't have strong feelings about these particular rules, but I do want consistency and to get rid of our minimal config ASAP.

Reply:
I am fine with the linter preferences in general. I don't do well remember arbitrary rules; I do much better understanding why. So, I was trying to determine the rationale which may be less complicated than I am :)